python/lsst/sims/maf/plots/twoDPlotters.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+from matplotlib import colors
+from matplotlib import ticker
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+from .plotHandler import BasePlotter
+from .perceptual_rainbow import makePRCmap
+
+__all__=['TwoDMap', 'VisitPairsHist']
+
+
+perceptual_rainbow = makePRCmap()
+
+class TwoDMap(BasePlotter):
+    def __init__(self):
+        self.plotType = 'TwoD'
+        self.objectPlotter = False
+        self.defaultPlotDict = {'title':None, 'xlabel':None, 'ylabel':None, 'label':None,
+                                'logScale':False, 'cbarFormat':None, 'cbarTitle': 'Count' ,
+                                'cmap':perceptual_rainbow,
+                                'percentileClip':None, 'colorMin':None, 'colorMax':None,
+                                'zp':None, 'normVal':None,
+                                'cbar_edge':True, 'nTicks':None, 'aspect':'auto'}
+
+    def __call__(self, metricValueIn, slicer, userPlotDict, fignum=None):
+
+        if 'Healpix' in slicer.slicerName:
+            self.defaultPlotDict['ylabel'] = 'Healpix ID'
+        elif 'Opsim' in slicer.slicerName:
+            self.defaultPlotDict['ylabel'] = 'Field ID'
+
+        plotDict = {}
+        plotDict.update(self.defaultPlotDict)
+        # Don't clobber with None
+        for key in userPlotDict.keys():
+            if userPlotDict[key] is not None:
+                plotDict[key] = userPlotDict[key]
+
+
+        if plotDict['logScale']:
+            norm = colors.LogNorm()
+        else:
+            norm = None
+
+        # Decide if we want all the zeropoint sillyness
+
+        metricValue = metricValueIn.copy()
```

Comment:
I'm not sure I understand the lines above -- you don't modify metricValue below, so if you don't use the zeropoint stuff, you don't need to copy it. 
If you do want the zeropoint stuff, it should be added.